src/test/java/seedu/address/logic/commands/AddClaimCommandTest.java

```diff
@@ -0,0 +1,139 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static seedu.address.logic.commands.CommandTestUtil.assertInsuranceCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalClients.getTypicalAddressBook;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_CLIENT;
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_CLIENT;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.client.Client;
+import seedu.address.model.client.exceptions.ClaimException;
+import seedu.address.model.client.exceptions.InsurancePlanException;
+import seedu.address.model.client.insurance.InsurancePlan;
+import seedu.address.model.client.insurance.InsurancePlansManager;
+import seedu.address.model.client.insurance.InvalidInsurancePlan;
+import seedu.address.model.client.insurance.claim.Claim;
+
+class AddClaimCommandTest {
+
+    public static final int VALID_INSURANCE_ID_FIRST = 0;
+    public static final int VALID_INSURANCE_ID_SECOND = 1;
+    public static final String VALID_CLAIM_ID_FIRST = "A1001";
+    public static final String VALID_CLAIM_ID_SECOND = "B1234";
+    public static final int VALID_CLAIM_AMOUNT_FIRST = 10000;
+    public static final int VALID_CLAIM_AMOUNT_SECOND = 20000;
+    public static final int INVALID_CLAIM_AMOUNT = -1;
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() throws
```

Comment:
i think it will be helpful to have javadoc comments for added clarity!

Reply:
Ok sure, I'll add it in 👍

Reply:
Added!